torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -285,18 +286,30 @@ auto handle_torch_function_no_python_arg_parser(
 
       // See https://github.com/pytorch/pytorch/issues/63767
       if (PyObject_FastGetAttrString(torch_function.ptr(), "__self__").is(arg) &&
-          torch_function.ptr() != torch::disabled_torch_function_impl()) {
-        TORCH_WARN("Defining your `__torch_function__` as a plain method is deprecated and ",
-                   "will be an error in future, please define it as a classmethod.");
+          torch_function.ptr() != torch::disabled_torch_function_impl() &&
+          !at::impl::PythonModeTLS::get_state()) {
```

Comment:
If you handle mode separately in the block above, you won't need this condition anymore. This condition is a bit finely balanced right now in any case, because you are indirectly saying that this `torch_function` is coming from a mode if there exists a mode, but if the mode itself returned NotImplemented you could have ended up in subsequent loops where this invariant is no longer true...